internal/core/window.rs

```diff
@@ -349,6 +361,91 @@ impl Window {
         }
     }
 
+    /// Take the focus_item out of this Window
+    ///
+    /// This sends the FocusOut event!
+    fn take_focus_item(self: Rc<Self>) -> ItemWeak {
+        let focus_item = self.as_ref().focus_item.take();
+
+        if let Some(focus_item_rc) = focus_item.upgrade() {
+            println!("Window: Taking existing focus_item.");
+            focus_item_rc.borrow().as_ref().focus_event(&crate::input::FocusEvent::FocusOut, &self);
+            focus_item_rc.downgrade()
+        } else {
+            println!("Window: Taking new focus_item.");
+            // No focus item was set: Try to move to the first item accepting focus:
```

Comment:
I don't understand why you need to do that in the function that sends the FocousOut event.

Reply:
But then you do things that shouldn't be done in the set_focus case.